subsys/net/ip/net_context.c

```diff
@@ -1726,8 +1775,21 @@ static int context_sendto(struct net_context *context,
 
 	} else if (IS_ENABLED(CONFIG_NET_IPV4) &&
 		   net_context_get_family(context) == AF_INET) {
```

Comment:
Is this the right place for the translation? This part of code is covering IPv4 sockets (AF_INET), I had an impression those should still cover IPv4 communications only? Shoudn't the IPv6 code (above) check for IPv4 mapped address?

Reply:
When the socket is accepted and if the incoming data is IPv4, then the accepted net_context family is set to IPv4 even if the listening one is IPv6 and the application socket is IPv6. It looks like a big mess if we try to handle IPv4 data in IPv6 path in other parts of the network stack. With these minimal changes, the echo sample application I modified works fine both in Zephyr and Linux.